src/commands.rs

```diff
@@ -662,6 +666,81 @@ pub fn json_set<M: Manager>(manager: M, ctx: &Context, args: Vec<RedisString>) -
     }
 }
 
+///
+/// JSON.MSET <key> <path> <json> [[<key> <path> <json>]...]
+///
+pub fn json_mset<M: Manager>(manager: M, ctx: &Context, args: Vec<RedisString>) -> RedisResult {
+    let mut args = args.into_iter().skip(1);
+
+    if args.len() < 3 {
+        return Err(RedisError::WrongArity);
+    }
+
+    // Collect all the actions from the args (redis_key, update_info, value)
+    let mut actions = Vec::new();
+    while let Ok(key) = args.next_arg() {
+        let mut redis_key = manager.open_key_write(ctx, key)?;
+
+        // Verify the key is a JSON type
+        let key_value = redis_key.get_value()?;
+
+        // Verify the path is valid and get all the update info
+        let path = Path::new(args.next_str()?);
+        let update_info = if path.get_path() == JSON_ROOT_PATH {
+            None
+        } else if let Some(value) = key_value {
+            Some(KeyValue::new(value).find_paths(path.get_path(), &SetOptions::None)?)
```

Comment:
* Since we find the paths on the first pass - we may fail updating the same key multiple times,

Which is possible in multi exec
e.g.,
This is OK
```
json.set a $ '{"x":[10,20]}'
json.set a '$.x[1]' '30'

json.get a $
"[{\"x\":[10,30]}]"
```

But this will fail, since before the `json.mset` command, `$.x[1]` does not exist:
```
json.mset a $ '{"x":[10,20]}' a '$.x[1]' '30'
```

* Since we modify the key on the second path, we may fail although the first pass does not detect it,
e.g., 
```
json.set a $ '{"x": {"y":[10,20], "z":[30,40]}}'

json.mset a $.x '{}' a '$.x.z[1]' '50'
```

The other alternative would be to do modifications in one pass, and if any fails, rollback, but that would be less efficient.
For multiple updates on the same key we can add a limitation that updates are incremental (and that path validations are not incremental?)

Reply:
Don't we have the issue in JSON.SET?

Reply:
@oshadmi see my last commit